Backend connection

djasmit opened the pull request:
Making Connections to the backend.
Currently does signup+login, delete and contact endpoints.
More details in Commits.

More work to be done in week 11. PR'ing now to have the link for evidence in week 10 task. Please don't merge this yet until description updated. Currently incomplete.

<img width="884" height="707" alt="image" src="https://github.com/user-attachments/assets/f6f6cff8-9e7f-4c4d-82cf-ca6354546bdd" />
<img width="975" height="295" alt="image" src="https://github.com/user-attachments/assets/ac812307-ed7c-4f06-9302-8f0ffd56d54b" />
<img width="985" height="490" alt="image" src="https://github.com/user-attachments/assets/7630a5a2-dc84-4d9e-a3eb-a83a1c6fcbef" />

djasmit committed:
Login + Signup

Modified the existing production code to allow login/signup pages to connect to backend.

Added .env for API_URL to connect to

Changes:

src/app/index.jsx (login page):
- production code now captures email/password input fields
- production code now generates valid form data
- production code no longer overrides content-type when making HTTP request

src/app/signup.jsx:
- production code no longer overrides content-type when making HTTP request

.gitignore:
- now ignores .env

.env.example:
- new file for .env value examples

djasmit committed:
Update Delete Confirmation

Updates the delete account endpoints.

Changes:

src/app/(settings)/deleteAccount.jsx (delete account page):
- "delete my account" button now captures form data
- form data is now sent to save_ta_message endpoint (terminate account message)

src/app/(deleteAccountConfirmation)/confirmIntent.jsx (delete confirmation page):
- production code no longer overrides content-type when making HTTP request
- replaced unfinished setDeleteSuccessful() call with routing to account deletion success page

Notes:
Additional reworks needed. Backend requires some work (delete message endpoint currently enforces JSON format unlike other endpoints)

djasmit committed:
Update contact.jsx

Updates the contact us messaging endpoint.
Changes:

src/app/(settings)/contact.jsx (contact us page):
- created a HandleContact() function that captures form data, bundles it with user.ID and posts as message to backend
- submit button now triggers handleContact() on press
- form data now uses enum indexing for category
- added user context to Contact () function

djasmit committed:
ENV Example Update

Updates the .env.example file to include the different required IPs for testing (Android Emulation is different)

djasmit committed:
Delete Logout User and Reason Grammar Correction

Updates the Delete endpoint
Changes:

src/app/(settings)/deleteAccount.jsx (delete page):
- updated grammer in options

src/app/(deleteAccountConfirmation)/confirmAccountDeletion.jsx (delete page):
- now uses AuthContext to log out user after deleting an account.

lperry022 commented on src/app/(settings)/contact.jsx:
Thanks for this update! The core logic looks solid and the API call is set up correctly. Before I can approve, could you:
- Fix the duplicate React import (merge into one line).
- Remove the leftover console.log statements that leak user data.
- Correct the typo "General Equiry" → "General Enquiry".

lperry022 commented on src/app/signup.jsx:
Thanks for the signup implementation! Before merging, can you:
- Remove debug console.log statements (they leak credentials and PII).
- Ensure navigation to /home happens only on successful signup (currently always redirects).

lperry022 commented on src/app/(settings)/deleteAccount.jsx:
```diff
@@ -49,6 +53,52 @@ const deleteAccount = () => {
+      reason: String(inData.reason),
+      message_body: String(inData.moreInfo ?? ""),
+    };
+    console.log(JSON.stringify(bodyData)); //remove this in PR
```
remove PII print

lperry022 commented on src/app/index.jsx:
```diff
@@ -18,40 +18,53 @@ import "@expo/metro-runtime";
+        break;
+      case 200:
+        const data = await response.json();
+        console.log(JSON.stringify(data)); //remove this in PR
```
remove PII print

lperry022 reviewed (changes requested):
@djasmit Thanks for the updates across the components! The overall structure looks solid, but before this can be merged a few clean-ups are needed:
- Contact.jsx:  remove debug console.log statements, and correct the "General Equiry" typo.
- deleteAccount.jsx: , strip out console.log statements,
- index.jsx (Login): strip out debug logs (especially credentials), and ensure redirect to /home only occurs after a successful login.
- signup.jsx: Remove debug logs, and make sure redirect to /home happens only on a successful signup.

Once these issues are addressed, this will be in great shape to approve. 👍

djasmit committed:
Merge branch 'Redback-Operations:main' into BackendConnection

djasmit committed:
Formatting and logout fixes

Just some touch ups to improve navigation and endpoint connections.

CHANGES:
Pages:
src/app/(deleteAccountConfirmation)/confirmIntent (Pre-Delete Confirmation):
- added note about 204 freezing error (requires backend work to resolve)
- added default case to HTTP response switch
- removed console logs

src/app/(deleteAccountConfirmation)/confirmAccountDeletion (Post-Delete Confirmation):
- now correctly logs out user

src/app/(settings)/contact.jsx (Contact Us Page):
- renamed category/categoryEnum to topic/topic Enum to match backend
- renamed topics to match backend version to correctly connect to backend
- removed console logs

src/app/(settings)/deleteAccount:
- removed console logs

src/app/index:
- removed console logs

src/app/signup:
- removed console logs

src/components/Setting.jsx
- logout now resets navigation
